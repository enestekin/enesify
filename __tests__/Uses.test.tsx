import { render, screen } from "@testing-library/react";
import Uses from "@/app/uses/page";

describe("Uses", () => {
  it("renders a single h1 heading", () => {
    render(<Uses />);
    const h1 = screen.getByRole("heading", { level: 1 });
    expect(h1).toBeInTheDocument();
    expect(h1).toHaveTextContent(/What Enes Uses/i);
  });

  it("renders multiple section headings", () => {
    render(<Uses />);
    const h2s = screen.getAllByRole("heading", { level: 2 });
    expect(h2s.length).toBeGreaterThanOrEqual(3);

    expect(screen.getByRole("heading", { name: /Tech/i })).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: /Desk Setup/i }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", {
        name: /Web Performance, Security & Monitoring/i,
      }),
    ).toBeInTheDocument();
  });

  it("labels each section with aria-labelledby", () => {
    render(<Uses />);
    expect(screen.getByRole("region", { name: "Tech" })).toBeInTheDocument();
    expect(
      screen.getByRole("region", { name: "Desk Setup" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("region", {
        name: "Web Performance, Security & Monitoring",
      }),
    ).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Uses />);
    expect(container).toMatchSnapshot();
  });
});
