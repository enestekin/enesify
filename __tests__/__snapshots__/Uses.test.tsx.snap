// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Uses Page renders correctly the page 1`] = `
<div>
  <main
    class="mx-auto max-w-4xl px-4 py-16 text-neutral-400"
  >
    <header
      class="mb-16"
    >
      <h1
        class="text-3xl text-white md:text-5xl"
      >
        What Enes Uses
      </h1>
      <p
        class="mt-2 text-2xl text-neutral-400"
      >
        The tools and technologies I rely on daily.
      </p>
    </header>
    <section
      aria-labelledby="tech-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="tech-heading"
      >
        Tech
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          React — My go-to UI library for building modern web apps.
        </li>
        <li>
          Next.js — The best React framework for production-grade apps.
        </li>
        <li>
          TypeScript — Strongly typed JavaScript that makes coding safer.
        </li>
        <li>
          TailwindCSS — Utility-first CSS for fast and clean styling.
        </li>
        <li>
          Prettier — Keeps my code clean and consistently formatted.
        </li>
        <li>
          React Testing Library — For testing UI the way users interact with it.
        </li>
        <li>
          Playwright — My favorite tool for end-to-end testing.
        </li>
        <li>
          Git & GitHub — Version control, collaboration, and CI/CD.
        </li>
        <li>
          NPM — The package manager powering all my projects.
        </li>
        <li>
          VSCode — My daily driver code editor.
        </li>
        <li>
          Notion — My second brain for notes and project planning.
        </li>
        <li>
          Discord — For staying connected with dev communities.
        </li>
        <li>
          GitHub Copilot / CodeWhisperer — AI pair programmers that boost productivity.
        </li>
        <li>
          MDX — Markdown with React components for rich content.
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="desk-setup-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="desk-setup-heading"
      >
        Desk Setup
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          M2 MacBook Pro 14-inch (2023) — My main development machine.
        </li>
        <li>
          Keychron K2 Keyboard — A solid mechanical keyboard.
        </li>
        <li>
          Apple Magic Keyboard — Slim and minimal, great for portability.
        </li>
        <li>
          Logitech MX Master Mouse — Ergonomic and perfect for long coding sessions.
        </li>
        <li>
          Acer Nitro ED320QXbi 31.5” Monitor
        </li>
        <li>
          JBL Tune 510BT Headphones
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="performance-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="performance-heading"
      >
        Web Performance, Security & Monitoring
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          Google Lighthouse — Auditing web performance and best practices.
        </li>
        <li>
          Sentry — Application monitoring and error tracking.
        </li>
        <li>
          Google Analytics — Tracking user behavior and performance.
        </li>
        <li>
          WebPageTest — In-depth web performance testing.
        </li>
        <li>
          PageSpeed Insights — Optimizing page speed.
        </li>
      </ul>
    </section>
  </main>
</div>
`;

exports[`Uses matches snapshot 1`] = `
<div>
  <main
    class="mx-auto max-w-4xl px-4 py-16 text-neutral-400"
  >
    <header
      class="mb-16"
    >
      <h1
        class="text-3xl text-white md:text-5xl"
      >
        What Enes Uses
      </h1>
      <p
        class="mt-2 text-2xl text-neutral-400"
      >
        The tools and technologies I rely on daily.
      </p>
    </header>
    <section
      aria-labelledby="tech-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="tech-heading"
      >
        Tech
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          React — My go-to UI library for building modern web apps.
        </li>
        <li>
          Next.js — The best React framework for production-grade apps.
        </li>
        <li>
          TypeScript — Strongly typed JavaScript that makes coding safer.
        </li>
        <li>
          TailwindCSS — Utility-first CSS for fast and clean styling.
        </li>
        <li>
          Prettier — Keeps my code clean and consistently formatted.
        </li>
        <li>
          React Testing Library — For testing UI the way users interact with it.
        </li>
        <li>
          Playwright — My favorite tool for end-to-end testing.
        </li>
        <li>
          Git & GitHub — Version control, collaboration, and CI/CD.
        </li>
        <li>
          NPM — The package manager powering all my projects.
        </li>
        <li>
          VSCode — My daily driver code editor.
        </li>
        <li>
          Notion — My second brain for notes and project planning.
        </li>
        <li>
          Discord — For staying connected with dev communities.
        </li>
        <li>
          GitHub Copilot / CodeWhisperer — AI pair programmers that boost productivity.
        </li>
        <li>
          MDX — Markdown with React components for rich content.
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="desk-setup-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="desk-setup-heading"
      >
        Desk Setup
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          M2 MacBook Pro 14-inch (2023) — My main development machine.
        </li>
        <li>
          Keychron K2 Keyboard — A solid mechanical keyboard.
        </li>
        <li>
          Apple Magic Keyboard — Slim and minimal, great for portability.
        </li>
        <li>
          Logitech MX Master Mouse — Ergonomic and perfect for long coding sessions.
        </li>
        <li>
          Acer Nitro ED320QXbi 31.5” Monitor
        </li>
        <li>
          JBL Tune 510BT Headphones
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="performance-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="performance-heading"
      >
        Web Performance, Security & Monitoring
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          Google Lighthouse — Auditing web performance and best practices.
        </li>
        <li>
          Sentry — Application monitoring and error tracking.
        </li>
        <li>
          Google Analytics — Tracking user behavior and performance.
        </li>
        <li>
          WebPageTest — In-depth web performance testing.
        </li>
        <li>
          PageSpeed Insights — Optimizing page speed.
        </li>
      </ul>
    </section>
  </main>
</div>
`;

exports[`UsesPage matches snapshot 1`] = `
<div>
  <main
    class="mx-auto max-w-4xl px-4 py-16 text-neutral-400"
  >
    <header
      class="mb-16"
    >
      <h1
        class="text-3xl text-white md:text-5xl"
      >
        What Enes Uses
      </h1>
      <p
        class="mt-2 text-2xl text-neutral-400"
      >
        The tools and technologies I rely on daily.
      </p>
    </header>
    <section
      aria-labelledby="tech-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="tech-heading"
      >
        Tech
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          React — My go-to UI library for building modern web apps.
        </li>
        <li>
          Next.js — The best React framework for production-grade apps.
        </li>
        <li>
          TypeScript — Strongly typed JavaScript that makes coding safer.
        </li>
        <li>
          TailwindCSS — Utility-first CSS for fast and clean styling.
        </li>
        <li>
          Prettier — Keeps my code clean and consistently formatted.
        </li>
        <li>
          React Testing Library — For testing UI the way users interact with it.
        </li>
        <li>
          Playwright — My favorite tool for end-to-end testing.
        </li>
        <li>
          Git & GitHub — Version control, collaboration, and CI/CD.
        </li>
        <li>
          NPM — The package manager powering all my projects.
        </li>
        <li>
          VSCode — My daily driver code editor.
        </li>
        <li>
          Notion — My second brain for notes and project planning.
        </li>
        <li>
          Discord — For staying connected with dev communities.
        </li>
        <li>
          GitHub Copilot / CodeWhisperer — AI pair programmers that boost productivity.
        </li>
        <li>
          MDX — Markdown with React components for rich content.
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="desk-setup-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="desk-setup-heading"
      >
        Desk Setup
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          M2 MacBook Pro 14-inch (2023) — My main development machine.
        </li>
        <li>
          Keychron K2 Keyboard — A solid mechanical keyboard.
        </li>
        <li>
          Apple Magic Keyboard — Slim and minimal, great for portability.
        </li>
        <li>
          Logitech MX Master Mouse — Ergonomic and perfect for long coding sessions.
        </li>
        <li>
          Acer Nitro ED320QXbi 31.5” Monitor
        </li>
        <li>
          JBL Tune 510BT Headphones
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="performance-heading"
      class="mb-16"
    >
      <h2
        class="text-2xl font-semibold text-white md:text-3xl"
        id="performance-heading"
      >
        Web Performance, Security & Monitoring
      </h2>
      <ul
        class="mt-6 ml-4 list-disc space-y-2 text-neutral-400 md:px-0"
      >
        <li>
          Google Lighthouse — Auditing web performance and best practices.
        </li>
        <li>
          Sentry — Application monitoring and error tracking.
        </li>
        <li>
          Google Analytics — Tracking user behavior and performance.
        </li>
        <li>
          WebPageTest — In-depth web performance testing.
        </li>
        <li>
          PageSpeed Insights — Optimizing page speed.
        </li>
      </ul>
    </section>
  </main>
</div>
`;
